include/fmt/core.h

```diff
@@ -292,6 +292,32 @@ template <typename T> using type_identity_t = typename type_identity<T>::type;
 template <typename T>
 using underlying_t = typename std::underlying_type<T>::type;
 
+template <typename...>
+struct disjunction;
+template <>
+struct disjunction<> : std::false_type {};
+template <typename P>
+struct disjunction<P> : P {};
+template <typename P1, typename P2>
+struct disjunction<P1, P2> : std::conditional<P1::value, P1, P2>::type {};
+template <typename P1, typename P2, typename P3, typename... Pn>
+struct disjunction<P1, P2, P3, Pn...> 
+  : std::conditional<P1::value, P1, disjunction<P2, P3, Pn...>>::type {
+};
```

Comment:
Why not use a simpler implementation like the one from [cppreference](https://en.cppreference.com/w/cpp/types/disjunction)?

```c++
template<class...> struct disjunction : [std::false_type](http://en.cppreference.com/w/cpp/types/integral_constant) { };
template<class B1> struct disjunction<B1> : B1 { };
template<class B1, class... Bn>
struct disjunction<B1, Bn...> 
    : [std::conditional_t](http://en.cppreference.com/w/cpp/types/conditional)<bool(B1::value), B1, disjunction<Bn...>>  { };
 ```

And the same for conjunction.

Reply:
I suggested the current implementation, because it is the same one that I committed for libstdc++ years ago and that worked for all known compilers regardless of completeness of variadic templates and minor rule changes in variadic templates. That being said, I guess the suggested alternative will work for all supported compilers. While adjusting this, we also can add a minor compile-conditional regarding `__cpp_lib_logical_traits` and take advantage of the STL variant in this case.

Reply:
> we also can add a minor compile-conditional regarding __cpp_lib_logical_traits and take advantage of the STL variant in this case.

I suggest keeping it simple and not introducing conditional compilation.

Reply:
OK.